cratedb_toolkit/testing/testcontainers/cratedb.py

```diff
@@ -51,44 +57,91 @@ class CrateDBContainer(KeepaliveContainer, DbContainer):
     CRATEDB_PASSWORD = os.environ.get("CRATEDB_PASSWORD", "")
     CRATEDB_DB = os.environ.get("CRATEDB_DB", "doc")
     KEEPALIVE = asbool(os.environ.get("CRATEDB_KEEPALIVE", os.environ.get("TC_KEEPALIVE", False)))
+    CMD_OPTS = {
+        "discovery.type": "single-node",
+        "node.attr.storage": "hot",
+        "path.repo": "/tmp/snapshots",
+    }
 
-    # TODO: Dual-port use with 4200+5432.
     def __init__(
         self,
         image: str = "crate/crate:nightly",
-        port: int = 4200,
+        ports: Optional[dict] = None,
         user: Optional[str] = None,
         password: Optional[str] = None,
         dbname: Optional[str] = None,
-        dialect: str = "crate",
+        cmd_opts: Optional[dict] = None,
         **kwargs,
     ) -> None:
+        """
+        :param image: docker hub image path with optional tag
+        :param ports: optional dict that maps a port inside the container to a port on the host machine;
+                      `None` as a map value generates a random port;
+                      Dicts are ordered. By convention the first key-val pair is designated to the HTTP interface.
+                      Example: {4200: None, 5432: 15432} - port 4200 inside the container will be mapped
+                      to a random port on the host, internal port 5432 for PSQL interface will be mapped
+                      to the 15432 port on the host.
+        :param user:  optional username to access the DB; if None, try `CRATEDB_USER` environment variable
+        :param password: optional password to access the DB; if None, try `CRATEDB_PASSWORD` environment variable
+        :param dbname: optional database name to access the DB; if None, try `CRATEDB_DB` environment variable
+        :param cmd_opts: an optional dict with CLI arguments to be passed to the DB entrypoint inside the container
+        :param kwargs: misc keyword arguments
+        """
         super().__init__(image=image, **kwargs)
 
-        self._name = "testcontainers-cratedb"  # -{os.getpid()}
-        self._command = "-Cdiscovery.type=single-node -Ccluster.routing.allocation.disk.threshold_enabled=false"
-        # TODO: Generalize by obtaining more_opts from caller.
-        self._command += " -Cnode.attr.storage=hot"
-        self._command += " -Cpath.repo=/tmp/snapshots"
+        self._name = "testcontainers-cratedb"
+
+        cmd_opts = cmd_opts if cmd_opts else {}
```

Comment:
Just a nit: I usually write this like that. Is it a good idea?

```suggestion
        cmd_opts = cmd_opts or {}
```

Reply:
Done